### PR TITLE
feat: add nani-account

### PR DESCRIPTION
**Pull request by nerderlyne**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [x] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

### Detailed summary
- Added `NaniAccountProviderConfig` type in `packages/accounts/src/nani-account/types.ts`.
- Exported `NaniAccount` and `createNaniAccountProvider` in `packages/accounts/src/index.ts`.
- Created `createNaniAccountProvider` function in `packages/accounts/src/nani-account/provider.ts`.
- Added `getDefaultNaniAccountFactoryAddress` utility function in `packages/accounts/src/nani-account/utils.ts`.
- Defined `NaniAccountFactoryConfigSchema` and `NaniAccountProviderConfigSchema` in `packages/accounts/src/nani-account/schema.ts`.
- Added `NaniAccountFactoryAbi` in `packages/accounts/src/nani-account/abis/NaniAccountFactoryAbi.ts`.
- Added `nani-account.ts` snippet in `site/snippets/nani-account.ts`.
- Added tests for `NaniAccount` in `packages/accounts/src/nani-account/__tests__/account.test.ts`.
- Updated documentation in `site/smart-accounts/accounts/nani-account.md`.

> The following files were skipped due to too many changes: `site/smart-accounts/accounts/nani-account.md`, `packages/accounts/src/nani-account/account.ts`, `packages/accounts/src/nani-account/abis/NaniAccountAbi.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/accounts/src/nani-account/account.ts [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+      args: [
+        calls.map((call: Exclude<UserOperationCallData, Hex>) => ({
+          ...call,
+          value: call.value ?? 0n,
+        })),
+      ],
+    });
+  }
+
+  async encodeExecuteDelegate(delegate: Address, data: Hex): Promise<Hex> {
```
Comment (moldy530) [inline packages/accounts/src/nani-account/account.ts:143]: should this be `static` as well?

File: packages/accounts/src/nani-account/abis/NaniAccountFactory.ts
Comment (moldy530) [inline packages/accounts/src/nani-account/abis/NaniAccountFactory.ts:1]: this is super nitpicky, but can we name this file `NaniAccountFactoryAbi.ts` 👀

**Review by moldy530 — COMMENTED** [review]
couple nits, but LGTM

File: packages/accounts/src/nani-account/abis/NaniAccountFactory.ts
Comment (nerderlyne) [reply packages/accounts/src/nani-account/abis/NaniAccountFactory.ts:1]: happy to comply with style ✊

**Review by moldy530 — APPROVED** [review]

**Pull request merged**